cmd/coco/reconcile/reconcile.go

```diff
@@ -0,0 +1,186 @@
+		reconcileBranchName: reconcileBranchName,
+	}, nil
+}
+
+func (r *ReconcileClient) Reconcile(dryRun bool) error {
+	return r.merge(dryRun)
+}
+
+func (r *ReconcileClient) merge(dryRun bool) error {
+	err := r.client.MergeBranches(r.target, r.source)
+	if err != nil {
+		if strings.Contains(err.Error(), "Merge conflict") {
+			return r.handleMergeConflict(dryRun)
+		} else {
+			return fmt.Errorf("failed to merge branches: %v", err)
+		}
+	}
+
+	if dryRun {
+		log.Sugar.Debug("No merge conflicts found (dry-run mode)")
+		return nil
+	}
+	log.Sugar.Info("Merged successfully")
+	return nil
+}
+
+func (r *ReconcileClient) handleMergeConflict(dryRun bool) error {
+	if dryRun {
+		return fmt.Errorf("merge conflicts detected")
+	}
+
+	reconcileBranch, err := r.client.GetBranch(r.reconcileBranchName)
+
+	if err != nil {
+		var resolved bool
+		resolved, err = r.handleExistingReconcileBranch(reconcileBranch)
+		if err != nil {
+			return err
+		}
+		if resolved {
+			return nil
+		}
+	}
+
+	return r.handleNewReconcileBranch()
+}
+
+func (r *ReconcileClient) handleExistingReconcileBranch(reconcileBranch *github.Branch) (bool, error) {
+	// Compare the latest target branch and reconcile branch
+	target, err := r.client.GetBranch(r.target)
+	if err != nil {
+		return false, fmt.Errorf("failed to get target branch: %v", err)
+	}
+	commits, err := r.client.CompareCommits(
+		reconcileBranch,
+		target)
+	if err != nil {
+		return false, fmt.Errorf("failed to compare commits: %v", err)
+	}
+	if commits.GetAheadBy() > 0 {
+		return r.handleTargetAhead()
+	} else {
+		//check mergability
+		return r.checkMergeability()
+	}
+}
+
+func (r *ReconcileClient) handleNewReconcileBranch() error {
+	// Create a new branch reconcile branch from target branch
+	target, err := r.client.GetBranchRef(r.target)
+	if err != nil {
+		return fmt.Errorf("Failed to get target branch reference: %v", err)
+	}
+	if err = r.client.CreateBranch(r.reconcileBranchName, target); err != nil {
+		return fmt.Errorf("Failed to create reconcile branch: %v", err)
+	}
+	log.Sugar.Debugf("Created new reconcile branch from %s", r.target)
+
+	pr, err := r.client.CreatePullRequest(r.source, r.reconcileBranchName)
+	if err != nil {
+		return fmt.Errorf("failed to create a draft PR: %v", err)
+	}
+	log.Sugar.Info("Draft pull request #%d created: %s\n", pr.GetNumber(), pr.GetHTMLURL())
+	return nil
+}
+
+func (r *ReconcileClient) checkMergeability() (bool, error) {
+	prs, err := r.client.ListPullRequests()
+	if err != nil {
+		return false, err
+	}
+	var pr *github.PullRequest
+	for _, p := range prs {
+		if p.Head.GetRef() == r.source && p.Base.GetRef() == r.reconcileBranchName {
+			pr = p
+			break
+		}
+	}
+	if pr != nil {
+		// check if the pull request is mergable
+		if pr.GetMergeable() {
+			// perform the merge
+			err = r.client.MergeBranches(r.target, r.reconcileBranchName)
+			if err != nil {
+				log.Sugar.Infof("Successfully merged %s to %s", r.reconcileBranchName, r.target)
```

Comment:
this looks wrong. 
success on err != nil?

Reply:
resolved, missed it while refactoring